lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -886,6 +965,36 @@ def _get_first_data_node(self) -> str | None:
 
         return None
 
+    def clean_all_relation_data(self) -> None:
+        """Clean up all relation data."""
+        if not self.charm.unit.is_leader():
+            return
+
+        for rel in self.charm.model.relations[self.relation_name]:
+            self._delete_rel_data(rel.id)
+
+    def _block_if_waiting_for_peer_cluster(self) -> bool:
```

Comment:
Similar comment, on whether this could not have been part of the error setting on the provider side to centralize error logic in a single method